citadel-internal-service/src/kernel/ext.rs

```diff
@@ -27,7 +27,9 @@ pub trait IOInterfaceExt: IOInterface {
         tokio::task::spawn(async move {
             let write_task = async move {
                 let response =
-                    InternalServiceResponse::ServiceConnectionAccepted(ServiceConnectionAccepted);
+                    InternalServiceResponse::ServiceConnectionAccepted(ServiceConnectionAccepted {
+                        request_id: None,
```

Comment:
Ideally, this request_id matches that of the initiated connection so we know which request this belongs to